src/web/middlewares/web_middleware_csrf.ml

```diff
@@ -1,6 +1,39 @@
-(* TODO
-https://docs.djangoproject.com/en/3.0/ref/csrf/#how-it-works *)
+let ( let* ) = Lwt.bind
 
-let m () =
-  let filter handler ctx = handler ctx in
-  Web_middleware_core.create ~name:"csrf" filter
+module CsrfService = struct
+  let add_to_ctx ctx ~token () =
+    let open Core_ctx in
+    let key : Token.t key = create_key () in
+    add key token ctx
+end
+
+module Make
+    (MessageService : Message.Sig.Service)
+    (TokenService : Token.Sig.SERVICE) =
+struct
+  let m () =
+    let filter handler ctx =
+      let open Lwt_result.Infix in
+      let* res =
+        Web_req.urlencoded ctx "csrf" >>= fun value ->
```

Comment:
@aronerben Here you are mixing `let*` with infix bind, please just use one style consistently. I prefer `let*` in Sihl.
So something like 
```ocaml
let* csrf = Web_req.urlencoded ctx "csrf" in
let* token = TokenService.find ctx ~value:csrf |> Lwt.map Result.ok_of_failwith in
...
```

Reply:
`>>=` comes from `Lwt_result` but the `let*` `Lwt`. So I would have to define different `let*` but as I take it, that's not possible because that's a syntax extension? Unless I inline and overwrite `let*` in the relevant code sections.

Reply:
Yes that is a bit painful because we don't consistently do error handling atm. Okay let's keep it for now.